pkg/resources/resources.go

```diff
@@ -522,7 +522,12 @@ func Apply(ctx context.Context, cli client.Client, in client.Object, opts ...cli
 
 	err = cli.Patch(ctx, u, client.Apply, opts...)
 	if err != nil {
-		return fmt.Errorf("unable to patch object %s: %w", u, err)
+		// Include GVK and namespace/name for debugging context without logging sensitive object data
+		objRef := fmt.Sprintf("%s %s", u.GroupVersionKind().String(), u.GetName())
+		if u.GetNamespace() != "" {
+			objRef = fmt.Sprintf("%s %s/%s", u.GroupVersionKind().String(), u.GetNamespace(), u.GetName())
+		}
+		return fmt.Errorf("unable to patch %s: %w", objRef, err)
```

Comment:
Since I see this pattern is used on several places, why not creating an helper function like 

```
func FormatObjectReference(obj *unstructured.Unstructured) string {
    gvk := obj.GroupVersionKind().String()
    objName := FormatUnstructuredName(obj)
    return fmt.Sprintf("%s %s", gvk, objName)
}
```

```suggestion
objRef := FormatObjectReference(u)
return fmt.Errorf("unable to patch %s: %w", objRef, err)
```